scripts/aws_replicate.py

```diff
@@ -0,0 +1,376 @@
+import os
+import subprocess
+import shlex
+import Queue
+import time
+import timeit
+import threading
+
+import json
+import boto3
+
+from cdislogging import get_logger
+from indexclient.client import IndexClient
+
+from errors import APIError
+from settings import PROJECT_MAP, INDEXD
+from utils import (
+    get_bucket_name,
+    get_fileinfo_list_from_manifest,
+    get_fileinfo_list_from_s3_manifest,
+    exec_files_grouping,
+)
+
+logger = get_logger("AWSReplication")
+
+global EXIT_FLAG
+EXIT_FLAG = 0
+
+global TOTAL_PROCESSED_FILES
+TOTAL_PROCESSED_FILES = 0
+
+mutexLock = threading.Lock()
```

Comment:
just put this in the `run` that runs the threads, and pass in a reference to the threads?